BBQ - Profile Role

AlexBeanie opened the pull request:
## Description
This PR adds the role attribute to the Profile class, allowing future authorization restrictions.

## New main implementation
- Endpoint `/api/profiles/<user_id>/role` added for role retrieval.
- Role attribute across class implementations and methods.
- Role Enum class, for limited role options

## Example
With an already registered user, type the route `/api/profiles/<user_id>/role`, specifying the user id in the GET request.

By default, the response should be like this:

<img width="147" height="79" alt="image" src="https://github.com/user-attachments/assets/f81735e6-8807-4a1b-b151-8e37a0e92fb6" />
 

## Considerations

- It is necessary to register a user for the endpoint to work, since the profile is created with default `USER` role.

AlexBeanie committed:
Add role attribute to profile

AlexBeanie committed:
Update tests to include Profile role

AlexBeanie committed:
Merge branch 'dev' into BBQ-RoleProfile

Pamber016 reviewed (changes requested):
In ProfileRepository you must make a change to the restore_profile function, as you forgot to also include the role there! It would be lines 116 on-wards:
```
    def restore_profile(self, user_id: int, favorites: list) -> Optional[Profile]:
        profile_data = {
            "user_id": user_id,
            "favorite_foods": favorites,
            "unfavorite_foods": [],
            "favorite_menus": [],
            # HERE <-----
        }
        restored_profile_data = self.profile_database.create_profile(profile_data)
        if restored_profile_data:
            return Profile(
                user_id=int(restored_profile_data["user_id"]),
                favorite_foods=(
                    restored_profile_data["favorite_foods"].split(";")
                    if restored_profile_data["favorite_foods"]
                    else []
                ),
                unfavorite_foods=(
                    restored_profile_data["unfavorite_foods"].split(";")
                    if restored_profile_data["unfavorite_foods"]
                    else []
                ),
                favorite_menus=(
                    restored_profile_data["favorite_menus"].split(";")
                    if restored_profile_data["favorite_menus"]
                    else []
                ),
            # AND HERE <----
            )
        return None
```
Otherwise it seems everything else works perfectly! well done 👍🏻

AlexBeanie committed:
Update restore_profile method in ProfileRepository

AlexBeanie commented:
> In ProfileRepository you must make a change to the restore_profile function, as you forgot to also include the role there! It would be lines 116 on-wards:
> 
> ```
>     def restore_profile(self, user_id: int, favorites: list) -> Optional[Profile]:
>         profile_data = {
>             "user_id": user_id,
>             "favorite_foods": favorites,
>             "unfavorite_foods": [],
>             "favorite_menus": [],
>             # HERE <-----
>         }
>         restored_profile_data = self.profile_database.create_profile(profile_data)
>         if restored_profile_data:
>             return Profile(
>                 user_id=int(restored_profile_data["user_id"]),
>                 favorite_foods=(
>                     restored_profile_data["favorite_foods"].split(";")
>                     if restored_profile_data["favorite_foods"]
>                     else []
>                 ),
>                 unfavorite_foods=(
>                     restored_profile_data["unfavorite_foods"].split(";")
>                     if restored_profile_data["unfavorite_foods"]
>                     else []
>                 ),
>                 favorite_menus=(
>                     restored_profile_data["favorite_menus"].split(";")
>                     if restored_profile_data["favorite_menus"]
>                     else []
>                 ),
>             # AND HERE <----
>             )
>         return None
> ```
> 
> Otherwise it seems everything else works perfectly! well done 👍🏻

Thanks, I just fixed it

AlexBeanie committed:
Merge branch 'dev' into BBQ-RoleProfile

AlexBeanie committed:
Merge branch 'dev' into BBQ-RoleProfile

AlexBeanie committed:
Merge branch 'dev' into BBQ-RoleProfile
